### ISSUE TITLE
Custom binding update not called if unwrap not used.

### ISSUE THREAD

**Issue by alvarotrigo** [knockout/knockout#2050]
The `update` callback seems only to be called when the `ko.unwrap` is being called. 
[Working callback](https://jsfiddle.net/4kua2v2p/2/)
[Not working callback](https://jsfiddle.net/4kua2v2p/1/)

This won't fire the update callback:

```javasscript
ko.bindingHandlers.slideVisible = {
    update: function(element, valueAccessor, allBindings) {
    	alert("changed");
      // First get the latest data that we're bound to
        var value = valueAccessor();
 
        // Next, whether or not the supplied model property is observable, get its current value
        var valueUnwrapped = ko.unwrap(value); 
    }
};


var viewModel = {
    giftWrap: ko.observable(true)
};
ko.applyBindings(viewModel);

````

I didn't see this behaviour specified in the docs.

**Comment by AlexanderMarkov** [comment]
Treat the update callback as a computed observable that will automatically call whenever any of dependencies change.

**Comment by alvarotrigo** [comment]
> Treat the update callback as a computed observable that will automatically call whenever any of dependencies change.

Yeah, that makes sense. but it would be nice to add it on the documentation so its behavior is clear.

**Comment by AlexanderMarkov** [comment]
What is not clear from the custom binding documentation?:)

> Knockout will call the update callback initially when the binding is applied to an element and track any dependencies (observables/computeds) that you access. When any of these dependencies change, the update callback will be called once again.

**Comment by alvarotrigo** [comment]
> when the binding is applied to an element and track any dependencies (observables/computeds) that you access.

"that you access" where? 

As I read it I though it would be called as long as the obserbable used in the bind changed.
For example: `data-bind="myCustom: demo"`  if `demo` changes, then the callback `update` would be called.

**Comment by alvarotrigo** [comment]
A very similar issue was treated [here](http://stackoverflow.com/questions/23878702/knockout-custom-binding-update-not-firing) and by looking at the up-votes it seems people find themselves wondering the same.

**Comment by chrisknoll** [comment]
if the model property 'demo' changes reference to something else (ie: model.demo(newValue) ) then the update binding will fire.  If you update some nested property of the demo observable it won't because unless your binding actually accessed some other observable inside the binding (that's what they mean by 'that you access'  as in 'access during the binding) then update will trigger because dependencies on calling update() will be created on any obserable evaluated during the binding.

How about try this: 
~~~ javascript
        var valueJS = ko.toJS(value); 
~~~

What this does is evaluates every field in your value included the nested objects (the entire value object graph). Any observable evalubated will create a dependency on the data-bind.  Any change to any observable property in vale will trigger the databindign.

**Comment by alvarotrigo** [comment]
> if the model property 'demo' changes reference to something else (ie: model.demo(newValue) ) then the update binding will fire.

Not if not used inside the `update` callback: https://jsfiddle.net/4kua2v2p/3/

**Comment by chrisknoll** [comment]
Right, sorry, i thought we were past that part:  just passing an observable into a binding doesn't set up any dependencies. Even accessing the valueAccessor() doesn't set up dependencies, as the 'valueAccessor' is a computed that returns the value that was passed into the binding (in this case, an observable). That's why valueAccessor() doesn't pick up a binding dependency, but valueAccessor()() does, and also ko.unwrapObservable(valueAccessor()) will set up a dependency.

My comment was directed at the case where you bind to an observable (maybe it's a 'draw scaterplot binding' and when some of the underlying data of the observable changes, the bining doesn't fire an update.  Only the observables that were accessed during the binding will be tracked.

**Comment by stalniy** [comment]
@alvarotrigo you can improve documentation if you think it's not good enough. Just create a pull request to `gh-pages` branch
